### PR TITLE
change hca user attributes validations

### PR DESCRIPTION
**Pull request by lihanli**
## Description of change
https://github.com/department-of-veterans-affairs/vets.gov-team/issues/17290
change hca user attributes validations to match the frontend

## Testing planned
test on dev

#### Applies to all PRs

- [x] Appropriate logging
- [x] Swagger docs have been updated, if applicable
- [x] Provide link to originating GitHub issue, or connected to it via ZenHub
- [x] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [x] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

File: lib/hca/enrollment_eligibility/status_matcher.rb [hunk 1]
```diff
@@ -131,29 +131,35 @@ def process_text_match(text_matches, ineligibility_reason)
         nil
       end
 
+      # rubocop:disable Metrics/CyclomaticComplexity
+      # rubocop:disable Metrics/PerceivedComplexity
```
Comment (hpjaj) [inline lib/hca/enrollment_eligibility/status_matcher.rb]: If you're interested, I saw this recently.  I believe you can bring these into one line like this:

```ruby
 # rubocop:disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
```

File: app/models/health_care_application.rb [hunk 2]
```diff
@@ -66,17 +66,25 @@ def process!
...
+    if loa3
+      parsed_status = HCA::EnrollmentEligibility::StatusMatcher.parse(
+        ee_data[:enrollment_status], ee_data[:ineligibility_reason]
+      )
+
+      ee_data.slice(
+        :application_date,
+        :enrollment_date,
+        :preferred_facility
+      ).merge(parsed_status: parsed_status)
```
Comment (hpjaj) [inline app/models/health_care_application.rb]: Would there be any value abstracting this logic out into a private method, named after its business purpose?  I'm not sure what `ee_data` is (I should probably dig in further), but something like:

```ruby
eligibility_data_for(x, y)
```
Or something to that effect?  

Perhaps even a new instance method on the `HCA::EnrollmentEligibility::Service` that you could call on `ee_data`?  Just some thoughts, you'd know better.

Comment (lihanli) [reply app/models/health_care_application.rb]: sorry, i accidentally started working on another issue in this ticket, i'll keep your comments in mind for the other issue

**Comment by lihanli** [comment]
based on standup discussion the gender field is going to be removed so i'll make another PR for that change

**Pull request closed without merging**